ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -73,23 +76,29 @@
     EXECUTE
   }
 
-  public StatementExecutor(final KsqlEngine ksqlEngine) {
+  public StatementExecutor(
+      final KsqlEngine ksqlEngine,
+      final HybridQueryIdGenerator hybridQueryIdGenerator
+  ) {
     this(
         ksqlEngine,
-        new StatementParser(ksqlEngine)
+        new StatementParser(ksqlEngine),
+            hybridQueryIdGenerator
     );
   }
 
   @VisibleForTesting
   StatementExecutor(
       final KsqlEngine ksqlEngine,
-      final StatementParser statementParser
+      final StatementParser statementParser,
+      final HybridQueryIdGenerator hybridQueryIdGenerator
   ) {
     Objects.requireNonNull(ksqlEngine, "ksqlEngine cannot be null.");
 
     this.ksqlEngine = ksqlEngine;
     this.statementParser = statementParser;
     this.statusStore = new ConcurrentHashMap<>();
+    this.queryIdGenerator = hybridQueryIdGenerator;
```

Comment:
nit: null check (can you do the others too while you're in here?)

```suggestion
    this.queryIdGenerator = Objects.requireNotNull(queryIdGenerator, "queryIdGenerator");
```